src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.rst

```diff
@@ -0,0 +1,111 @@
+
+
+Detailed Module Description
+---------------------------
+A detailed mathematical derivation of the equations implemented in this module can be found in `R. Calaon, L. Kiner, C. Allard and H. Schaub, "Momentum Management of a Spacecraft equipped with a Dual-Gimballed Electric Thruster"  <http://hanspeterschaub.info/Papers/Calaon2023a.pdf>`__.
+
+This module computes a direction cosine matrix :math:`[\mathcal{FM}]` that describes the rotation between the platform frame :math:`\mathcal{F}` and the mount frame :math:`\mathcal{M}`. To be compliant with the constraint in the motion of the platform, i.e. the dual gimbal, such frame must have a zero in the element (2,1). When such condition is met, the reference angles computed from the D.C.M. allow to align the thruster through the system's center of mass. The input parameters for this module allow to specify offsets between the origin :math:`M` of the hub-fixed mount frame :math:`\mathcal{M}` and the origin :math:`F` of the platform-fixed frame :math:`\mathcal{F}`, the application point of the thruster force in the :math:`\mathcal{F}` frame, and the direction, in :math:`\mathcal{F}`-frame coordinates, of the thrust vector.
+
+When the optional input messages ``rwConfigDataInMsg`` and ``rwSpeedsInMsg`` the user can specify an input parameter ``K``, which is the proportional gain of a control gain that computes an offset with respect to the center of mass: this allows for the thruster to apply a torque on the system that dumps the momentum accumulated on the wheels. Such control law has the expression:
+
+.. math:: 
+    \boldsymbol{d} = \frac{\kappa}{t^2} (\boldsymbol{t} \times \boldsymbol{H}_w)
+
+where :math:`\boldsymbol{H}_w` is the momentum on the wheels.
+
+
+Module Assumptions and Limitations
+----------------------------------
+As pointed out in the paper referenced above, it is not always guaranteed that a direction cosine matrix exists, that can satisfy both the pointing requirement on the thrust direction and the kinematic constraint on the dual-gimballed platform. When a solution does not exist, a minimum problem is solved to compute the closest constraint-incompliant D.C.M. The tip and tilt referemce angles :math:`\nu_{1R}` and :math:`\nu_{2R}` are extracted from the final D.C.M. according to:
+
+.. math::
+    \begin{align}
+        \nu_{1R} &= \arctan \left( \frac{f_{23}}{f_{22}} \right) & 
+        \nu_{2R} &= \arctan \left( \frac{f_{31}}{f_{11}} \right)
+    \end{align}
+
+without checking whether the D.C.M. :math:`[\mathcal{FM}]` is constraint compliant. As a result, the angles :math:`\nu_{1R}` and :math:`\nu_{2R}` produce a constraint compliant reference, which however might not align the thruster with the desired point in the hub.
+
+
+User Guide
+----------
+The required module configuration is::
+schaubh marked this conversation as resolved.
+Show resolved
+
+    platformConfig = thrusterPlatformReference.thrusterPlatformReferenceConfig()
```

Comment:
this code snippets should be rendered as a python code snippet.  Right now it just makes a block of text.

Reply:
Fixed it.